src/index.ts

```diff
@@ -86,37 +65,38 @@ function fontawesomeSubset(subset: SubsetOption, outputDir: string, options: Fon
         }
 
         const fontFamily = key as keyof typeof fontMap;
-        const svgFileName = fontMap[fontFamily];
-        const svgFilePath = `node_modules/@fortawesome/fontawesome-${options.package}/webfonts/${svgFileName}.svg`;
+        const fontFileName = fontMap[fontFamily];
+        const fontFilePath = resolve(fontFiles, `./${fontFileName}.ttf`);
 
-        if (!existsSync(svgFilePath)) {
-            console.warn(`Unable to find SVG font file for requested font style '${fontFamily}'. Skipping.`);
+        if (!existsSync(require.resolve(fontFilePath))) {
+            console.warn(`Unable to find font file for requested font style '${fontFamily}'. Skipping.`);
             continue;
         }
 
-        const svgFile = readFileSync(svgFilePath).toString();
-        const glyphsToRemove = findGlyphsToRemove(svgFile, fontFamily, icons);
-        const svgContentsNew = svgFile.replace(new RegExp(`(<glyph glyph-name="(${glyphsToRemove.join("|")})".*?\\/>)`, "gms"), "").replace(/>\s+</gms, "><");
-        const ttfUtils = svg2ttf(svgContentsNew, {
-            fullname: "FontAwesome " + fontFamily,
-            familyname: "FontAwesome",
-            subfamilyname: fontFamily,
-            ts: 0 // Manually specify empty timestamp for deterministic output
-            // Casting these as @types are out of date for svg2ttf
-        } as FontOptions);
-        const ttf = Buffer.from(ttfUtils.buffer);
+        // Pull unicode characters from fontawesome yml, aggregating into array
+        const iconMeta: Record<GlyphName, { unicode: string }> = yaml.parse(readFileSync(fontMeta, "utf8"));
+        let unicodeCharacters: string[] = [];
+        for (const icon of icons) {
+            if (!(icon in iconMeta)) {
+                console.warn(`Icon '${icon}' is not found in font metadata. Skipping.`);
```

Comment:
One thing to be aware of (which might be OK) is that this does not support icon aliases. The alias names are in the YAML data so it is possible. I don't know if alias names was supported in the previous version for 5.x.

Reply:
Think I'll push that one to another PR / minor update. I think it may be possible with the `shims.yml`, but it only appears to support a subset of the old ones anyway. Could be deprecated.